### PR TITLE
feat: workflow qc

### PR DESCRIPTION
**Pull request by rannick**
### This PR:

Changed: 
* QC-workflow for panel cases to be run only with `balsamic-qc`
* `get_snakefile` function takes the argument `analysis_workflow` to trigger the QC workflow when necessary

Removed: 
* Removed `qc_panel` config in favor of standard config
* Removed cli `--analysis_type` for `balsamic report deliver` command and `balsamic run analysis`
* Removed `analysis_type`: `qc_panel` and replace the trigger for QC workflow by `analysis_workflow`: `balsamic-qc`

Fixed:
* GENOME_VERSION set to the different genome_version options, not always hg19 any more

_Test command line_
```
balsamic config case   --tumor tests/test_data/fastq/S1_R_1.fastq.gz   --normal tests/test_data/fastq/S2_R_1.fastq.gz   --case-id demo_run_balsamic   --analysis-dir cancer-test-data/demo   --panel-bed tests/test_data/references/panel/panel.bed --balsamic-cache /home/proj/development/cancer/balsamic_cache/ --genome-version hg19 --analysis-workflow balsamic-qc

balsamic run analysis --sample-config cancer-test-data/demo/demo_run_balsamic/demo_run_balsamic.json --profile slurm --qos low --account development --run-analysis
```
 
### Review and tests: 
- [ ] Tests pass
- [ ] Code review
- [ ] New code is executed and covered by tests, and test approve

### CODE REVIEW

**Comment by rannick** [comment]
The codecov drop in coverage seems to be due to the file config/qc.py being deleted with the corresponding tests. I think it influences the ration and hits but codecov cannot display the problematic lines because they do not exist any more

File: BALSAMIC/workflows/balsamic.smk [hunk 1]
```diff
@@ -106,10 +106,15 @@ if not Path(config["SENTIEON_EXEC"]).exists():
     LOG.error("Sentieon executable not found {}".format(Path(config["SENTIEON_EXEC"]).as_posix()))
     raise BalsamicError
 
-# Add reference assembly if not defined for backward compatibility
-if 'genome_version' not in config["reference"]:
-    GENOME_VERSION = 'hg19' ## if hg19 convention works, replace accordingly
-    LOG.info('Genome version was not found in config. Setting it to %s', GENOME_VERSION)
+if "hg38" in config["reference"]["reference_genome"]:
+    config["reference"]["genome_version"] = "hg38"
+elif "canfam3" in config["reference"]["reference_genome"]:
```
Comment (ashwini06) [inline BALSAMIC/workflows/balsamic.smk:111]: Will it make sense to run the balsamic workflow with dog genomes? As for variant calling, we don't have certain SNP databases which we currently have for humans.

File: CHANGELOG.rst [hunk 2]
```diff
@@ -11,14 +11,29 @@ Changed:
 
 * UMI-workflow for panel cases to be run only with `balsamic-umi` flag https://github.com/Clinical-Genomics/BALSAMIC/issues/896
 * Update `codecov` action version to @v2 https://github.com/Clinical-Genomics/BALSAMIC/pull/941
+* QC-workflow for panel cases to be run only with `balsamic-qc`
```
Comment (ashwini06) [inline CHANGELOG.rst:14]: Please add the PR/issue url links to all the new items in the  changelog

**Review by ivadym — COMMENTED** [review]
👍 

Could you please update which files should be stored and/or delivered with `balsamic-qc` (Delivery tag column): https://docs.google.com/spreadsheets/d/1QGJpdaeKLZGWCYseEXja9vOcCeVMNwQo/edit#gid=864744071

File: BALSAMIC/workflows/balsamic.smk (hunk 1 above)
Comment (rannick) [reply BALSAMIC/workflows/balsamic.smk:111]: How about throwing an error when canfam3 is used in combination with balsamic main workflow?

Comment (rannick) [reply BALSAMIC/workflows/balsamic.smk:111]: Suggestion in 84441c148d0bf4c3d2217bf9e902dcffc2665a76

File: CHANGELOG.rst (hunk 2 above)
Comment (rannick) [reply CHANGELOG.rst]: Adressed in af718d242d52a1b1bb6609987be6bad529b39ca6

**Comment by rannick** [comment]
> Added quality metrics in [d4e0d3c](https://github.com/Clinical-Genomics/BALSAMIC/commit/d4e0d3c53b36d04c85fd625cfe0803f3f84cb4f6)

The qc-metrics rule fails because no `filtered.pass.stats` file exists in the balsamic-qc. The rule needs to be changed prior to merging this PR.

**Review by ashwini06 — APPROVED** [review]
Looks good 💯

**Pull request merged**